store/sqlstore/post_store.go

```diff
@@ -555,8 +555,8 @@ func (s *SqlPostStore) GetPostsSince(channelId string, time int64, limit int, al
 			    WHERE
 			        UpdateAt > :Time
 			            AND ChannelId = :ChannelId
-			    LIMIT :Limit) temp_tab))
-			ORDER BY CreateAt DESC`,
+				LIMIT 1000) temp_tab))
+			ORDER BY CreateAt ASC LIMIT :Limit`,
```

Comment:
[HH comment](https://github.com/mattermost/mattermost-server/pull/10861#discussion_r286047410): Why change the order from descending to ascending?

Reply:
__My original thought is that:__ I changed it 'cause I think the old implementation of `GetPostsSince` does not sound accurate.  The existing limit of 1000 seems to mean return "all" posts from a certain time to new post.  We don't have any issue with it being "all" since our client usually don't ask for a very far "since" and as a result it returns less than 1000 posts.

However with unread API, it only gets limited number of posts around last unread (or a period that the channel was last viewed) and returns a number of post (say 60) from such period (old) to recent post).  If we keep DESC and put limit of 60 on it, it will return 60 posts starting from recent to old, and as a result it will going to have missing messages in between.

However, upon looking closer on the changes I've made, I realized that I should have not touch the GetPostSince and should just stick on GetPostsAfter.  I'll revise the PR...

Reply:
Are you still making that change to use `GetPostsAfter` for `GetPostsForChannelAroundLastUnread`?

Reply:
Yes, I did it here - https://github.com/mattermost/mattermost-server/pull/11039/files#diff-317b12c12f71e311f13813107f4d8550R807

Reply:
Are these changes to `GetPostsSince` still needed then? I'm just leery about changing this function since it already works a bit unexpectedly

Reply:
No longer needed for unread API.  I've reverted all changes to GetPostSince in reference to master.